### ISSUE TITLE
App Crash while checking State in OAuthMobile.swift

### ISSUE THREAD

**Issue by sagarmilani** [dropbox/SwiftyDropbox#202]
I am developing an iOS App which is using file download from Dropbox. On one of my devices I have updated to latest version of Dropbox App and I am trying to authenticate using that app but due to some reason my app crash on line **"if state.count == 2 && state[0] == "oauth2" && state[1] == nonce!"** which is in OAuthMobile.swift class.Also I have attached Crash Report for same. 
[CrashReports.txt](https://github.com/dropbox/SwiftyDropbox/files/1638628/CrashReports.txt)

**Comment by greg-db** [comment]
Thanks for the report. This isn't reproducing for me, so please let me know:
- the version number of the SwiftyDropbox SDK you're using
- the version number of the official Dropbox iOS app you have installed
- what, if any, error/output you get in the Xcode console when this occurs

Thanks in advance!

**Comment by sagarmilani** [comment]
After some research on the above Issue we found out that this is not actually the issue related to DropBox directly. But will share our findings so that if at later stage if anybody else get the same issue he’s time should not be wasted. 

**So here is what we find out:** 

At our end this crash was produced only for **Devices having multiple version’s installed of same App**. That means when our app was redirected to dropbox app for Authentication while returning back it launches the same app with different version installed in device which gives crash at auth state in dropbox.
So Closing this issue for now.
